web/modules/custom/dpl_login/src/Controller/DplLoginController.php

```diff
@@ -149,14 +155,26 @@ public function loginHandler(Request $request): TrustedRedirectResponse {
   /**
    * Get access token. If user is not registered, get unregistered user token.
    *
+   * @return mixed[]|null
+   *   Access token or NULL if not found.
+   *
    * @todo Should be moved to separate service that can be used througut the application.
    */
-  protected function getAccessToken(): ?AccessToken {
+  protected function getAccessToken(): ?array {
     if ($access_token = $this->unregisteredUserTokensProvider->getAccessToken()) {
-      return $access_token;
+
+      $tokenInfo = [
+        'token' => $access_token,
+        'type' => 'unregistered-user',
+      ];
+      return $tokenInfo;
     }
     if ($access_token = $this->userTokensProvider->getAccessToken()) {
-      return $access_token;
+      $tokenInfo = [
+        'token' => $access_token,
+        'type' => 'user',
+      ];
+      return $tokenInfo;
     }
```

Comment:
I think we can do a more pretty solution here.
As far as I see it we could do one of two things:
* Return the whole acces token object in this function and set the return type to the interface
* Implement a property on the token classes that says which type it is

I think I am most fan of the first solution. Then we can assert the type of the token by looking at which class the token is.